aya/src/bpf.rs

```diff
@@ -638,14 +638,43 @@ impl<'a> BpfLoader<'a> {
                 (name, program)
             })
             .collect();
-        let maps = maps
-            .drain()
-            .map(|(name, map)| (name, MapLock::new(map)))
-            .collect();
-        Ok(Bpf { maps, programs })
+        let maps: Result<HashMap<String, Map>, BpfError> = maps.drain().map(parse_map).collect();
```

Comment:
I don't think the explicit type is needed here?

Reply:
Hrm? 

```
error[E0282]: type annotations needed
   --> aya/src/bpf.rs:641:13
    |
641 |         let maps = maps.drain().map(parse_map).collect();
    |             ^^^^
    |
help: consider giving `maps` an explicit type
    |
641 |         let maps: _ = maps.drain().map(parse_map).collect();
    |                 +++
```

Reply:
ah yep sorry parse_map returns Result, so the type annotation here is needed to turn `Vec<Result<_>>` into `Result<Vec<_>>`